### ISSUE TITLE
Find a way to reference NixOS option in CommonMark

### ISSUE THREAD

**Issue by Ma27** [NixOS/nixpkgs#127813]
In #127134 the release-notes of older NixOS releases were migrated to Markdown as part of [RFC0072](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md). As we realized in a brief [conversation after the merge](https://github.com/NixOS/nixpkgs/pull/127134#discussion_r656341823), the CommonMark approach appears to have the downside that you can't really link NixOS options in codeblocks.

For instance, in DocBook one would do something like `<programlisting>{ <xref linkend="opt-services.fooo.enable" /> = true; }</programlisting>` and the option would be a link that "guides" the reader to the documentation of the option itself (where relevant things such as the type or further information in the option's description can be found).

Personally I really liked that and would be very upset to see that go as this is (well, for me at least - can't speak for anyone else) more intuitive than having to search for an option I just read about in the manual.

Unless there's a way to either implement this somehow trivially in CommonMark (or - well - file a patch there) I can currently think of two other options:

* As @blaggacao has pointed out correctly, this is pretty inconsistently used across the codebase, so perhaps I'm the only one who cares. If that's the case, we may want to leave it as-is.
* As @ryantm suggested, we could also automatically add links to the options used in a code-snippet (IIUC).

Apart from the folks I CCed above, I'd also like to ping @mboes who authored RFC72 and perhaps has an opinions (or some insights) on this.

**Comment by ryantm** [comment]
cc @jtojnar too

**Comment by ryantm** [comment]
My preference would be to do these links as a post-processing step (either Javascript or an HTML post-processing step). People wouldn't have to remember to add them, and maybe we could even make it smart enough to handle the nested ones.

**Comment by blaggacao** [comment]
If a post-processor encounters a `nix` codefence, a `nix instantiate` could yield the attribute tree and post-process thos links onto the end result globally.

If it's globally used (and it seams feasible), the usefulness of this is greatly enhanced, so :+1:

**Comment by rnhmjoj** [comment]
Post-processing is a good idea: having to type two times the option full path is quite painful.
The how likely depends on the specific implementation of markdown: has one been chosen yet?

**Comment by ryantm** [comment]
> The how likely depends on the specific implementation of markdown

I don't think it needs to be done at this layer. It can be a tool that takes HTML and returns HTML, for example.

> has one been chosen yet?

I'm developing a tool called [mmdoc](https://github.com/ryantm/mmdoc) that I'd like to use, but no, one has not been decided yet. Here is my [implementation of using mmdoc for the Nixpkgs Manual](https://github.com/NixOS/nixpkgs/pull/108063).

**Comment by Ma27** [comment]
Still relevant IMHO.
